block/manager.go

```diff
@@ -285,6 +287,15 @@ func (m *Manager) UpdateTargetHeight(h uint64) {
 			break
 		}
 	}
+
+	// updating LastGossipedHeight to avoid gossiping long queue of blocks when sequencer rotation
+	if m.State.LastGossipedHeight < h {
```

Comment:
2 issues I have with that:
1. the function is called update target height, but it actually also updates last gossiped height
2. the param is called last gossiped height for a full node, but it's tracking actually a potential sequencer only metric even if you're not a sequencer which may never be relevant for you. so maybe we need to make it a bit more make sense cause it feels hacky now (as if one would see that he would think it's the last height the full node got from gossiping but it's not)
3. we call the store save state on each iteration vs just updating it once. not sure why we need to iterate vs just saving h? it's just overriding in a loop.

Reply:
in order to avoid these issues, i refactored the pr and i implemented a new and simpler approach.
the new approach relies the block is added to blocksync after is produced but not on gossip. 
this way if the block is not gossiped because a panic, it will be available through blocksync, and full-nodes will be able to sync via p2p anyway